src/lib/network_pool/snark_pool.ml

```diff
@@ -148,6 +151,15 @@ struct
                  ]
                :: acc ) )
 
+      let all_completed_checked_work t =
+        Map.fold ~init:[] !(t.snark_tables).all
```

Comment:
1. You don't need to convert it to `Checked.t` (commented above). 
2. As discussed, this is a lot of snark work- all completed work referenced by the most recent 290 blocks in the best chain and any short forks within that range. In the max TPS case that'd be ~128*290*2 (assuming there are 2 blocks per slot, one canonical and one orphaned). Limiting helps but doesn't ensure that immediately relevant work is obtained. I think querying work referenced by recent, say 5 blocks in the best chain might be preferable.